lib/parsec/util.py

```diff
@@ -15,123 +15,140 @@
-                elif val is None:
-                    v = none_str
+    stack = [("", cfg, level, indent)]
+    while stack:
+        key_i, cfg_i, level_i, indent_i = stack.pop()
+        spacer = " " * 4 * (indent_i - 1)
+        if isinstance(cfg_i, dict):
+            if not cfg_i and none_str is None:
+                # Don't print empty sections if none_str is None. This does not
+                # handle sections with no items printed because the values of
+                # all items are empty or None.
+                continue
+            if key_i and level_i:
+                # Print heading
+                handle.write("%s%s%s%s%s\n" % (
+                    prefix, spacer, '[' * level_i, str(key_i), ']' * level_i))
+            # Nested sections are printed after normal settings
+            subsections = []
+            values = []
+            for key, item in cfg_i.items():
+                if isinstance(item, dict):
+                    subsections.append((key, item, level_i + 1, indent_i + 1))
                 else:
-                    v = str(val)
-                # print "key = val"
-                handle.write("%s%s%s = %s\n" % (prefix, spacer, str(key), v))
-
-        for key,val in delayed:
-            # print heading
-            #if val != None:
-            handle.write("%s%s%s%s%s\n" % (prefix, spacer, '['*(level+1), str(key), ']'*(level+1)))
-            # recurse into section
-            printcfg(val, level+1, indent+1, prefix, none_str, handle)
-
-def replicate( target, source ):
-    """
-    Replicate source *into* target. Source elements need not exist in
-    target already, so source overrides common elements in target and
-    otherwise adds elements to it.
+                    values.append((key, item, level_i + 1, indent_i + 1))
+
+            stack += reversed(subsections)
+            stack += reversed(values)
+        else:
+            key = ""
+            if key_i:
+                key = "%s = " % key_i
+            if cfg_i is None:
+                value = none_str
+            elif isinstance(cfg_i, list):
+                value = listjoin(cfg_i, none_str)
+            elif "\n" in str(cfg_i) and key:
+                value = '"""\n%s\n"""' % (cfg_i)
+            else:
+                value = str(cfg_i)
+            if value is not None:
+                handle.write("%s%s%s%s\n" % (prefix, spacer, key, value))
+
+
+def replicate(target, source):
+    """Replicate source *into* target.
+
+    Source elements need not exist in target already, so source overrides
+    common elements in target and otherwise adds elements to it.
     """
     if not source:
         target = OrderedDictWithDefaults()
         return
     if hasattr(source, "defaults_"):
         target.defaults_ = pdeepcopy(source.defaults_)
-    for key,val in source.items():
-        if isinstance( val, dict ):
+    for key, val in source.items():
+        if isinstance(val, dict):
```

Comment:
I guess we're not applying PEP8 to `lib/parsec`.  OK, we sure aren't (just checked).  My fault.